numpy/array_api/tests/test_asarray.py

```diff
@@ -0,0 +1,24 @@
+import numpy as np
+
+
+def test_fast_return():
+    """"""
+    a = np.array([1, 2, 3], dtype='i')
+    assert np.asarray(a) is a
+    assert np.asarray(a, dtype='i') is a
+    # This may produce a new view or a copy, but is never the same object.
+    assert np.asarray(a, dtype='l') is not a
```

Comment:
Note that this is only a good test on windows; it's `l` and `q` that clash on Linux, where l`long` is 64 bit

Reply:
Good point. I have added more permutations to this block so that at least one check would not pass without this patch.